In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from pandas_profiling import ProfileReport
import pickle
import warnings
warnings.simplefilter("ignore")

In [3]:
df = pd.read_csv("Advertising.csv")

In [4]:
df.head()

TV  Radio  Newspaper  Sales
0  230.1   37.8       69.2   22.1
1   44.5   39.3       45.1   10.4
2   17.2   45.9       69.3   12.0
3  151.5   41.3       58.5   16.5
4  180.8   10.8       58.4   17.9

In [5]:
df.tail()

TV  Radio  Newspaper  Sales
195   38.2    3.7       13.8    7.6
196   94.2    4.9        8.1   14.0
197  177.0    9.3        6.4   14.8
198  283.6   42.0       66.2   25.5
199  232.1    8.6        8.7   18.4

In [7]:
df_x = df[['TV','Radio','Newspaper']]
df_x

TV  Radio  Newspaper
0    230.1   37.8       69.2
1     44.5   39.3       45.1
2     17.2   45.9       69.3
3    151.5   41.3       58.5
4    180.8   10.8       58.4
..     ...    ...        ...
195   38.2    3.7       13.8
196   94.2    4.9        8.1
197  177.0    9.3        6.4
198  283.6   42.0       66.2
199  232.1    8.6        8.7

[200 rows x 3 columns]

In [10]:
y = df[['Sales']]
y.head()

Sales
0   22.1
1   10.4
2   12.0
3   16.5
4   17.9

In [12]:
from sklearn.linear_model import LinearRegression

In [15]:
model = LinearRegression()
model.fit(df_x,y )

LinearRegression()

In [16]:
model.intercept_

array([4.62512408])

In [17]:
model.coef_

array([[0.05444578, 0.10700123, 0.00033566]])

From here we understand that, investment on newspaper is benefitial. Not much affect on sales because of Newspaper...

Here, feature selection come into the picture...

Equation is;

y = 0.0544TV + 0.107Radio + 0.000335Newspaper + 4.625

In [20]:
model.score(df_x,y)        # This give the accuracy or confidence of the model 

0.9025912899684558

So, according to the result of linear regresion and multiple regression, here we get the good accuracy ac compared to result of linear regression.